### PR TITLE
frontend - fix: add rocket-loader param to elementor-frontend script tag.

### PR DESCRIPTION
**Pull request by nuritsha**

### CODE REVIEW

File: includes/frontend.php [hunk 1]
```diff
@@ -499,6 +503,15 @@ public function enqueue_scripts() {
...
 
+	public function elementor_frontend_rocket_loader_filter( $tag, $handler, $src ) {
+
+		if ( 'elementor-frontend' === $handler ) {
+			return str_replace( "<script type='text/javascript'", "<script type='text/javascript' data-cfasync='false'", $tag );
+		}
+
+		return $tag;
+	}
+
```
Comment (matipojo) [inline includes/frontend.php]: \Elementor\Core\Base\App::print_config
```
final protected function print_config() {
		$name = $this->get_name();

		echo '<script data-cfasync="false">' . "\n";
		echo 'var elementor' . ucfirst( $name ) . 'Config = ' . wp_json_encode( $this->get_settings() + $this->get_components_config() ) . ";\n";
		echo "</script>\n";
	}
```

**Comment by bainternet** [comment]
Still missing elementorFrontendConfig

**Comment by bainternet** [comment]
here also:
https://github.com/elementor/elementor/blob/fe5eb3949a9ea8b72e84616468614fc1a77e9262/includes/editor.php#L712-L724

File: includes/frontend.php [hunk 2]
```diff
@@ -499,6 +503,15 @@ public function enqueue_scripts() {
 		do_action( 'elementor/frontend/after_enqueue_scripts' );
 	}
 
+	public function elementor_frontend_rocket_loader_filter( $tag, $handler, $src ) {
+
+		if ( 'elementor-frontend' === $handler ) {
```
Comment (bainternet) [inline includes/frontend.php]: if ( isset( $_REQUEST['elementor-preview'] ) || ( isset( $_REQUEST['action'] ) && 'elementor' === $_REQUEST['action'] ) ) {

Comment (matipojo) [reply includes/frontend.php]: Plugin::instance()->preview->is_preview_mode();

File: includes/frontend.php [hunk 3]
```diff
@@ -505,8 +505,8 @@ public function enqueue_scripts() {
 
 	public function elementor_frontend_rocket_loader_filter( $tag, $handler, $src ) {
 
-		if ( 'elementor-frontend' === $handler ) {
-			return str_replace( "<script type='text/javascript'", "<script type='text/javascript' data-cfasync='false'", $tag );
+		if ( isset( $_REQUEST['elementor-preview'] ) || ( isset( $_REQUEST['action'] ) && 'elementor' === $_REQUEST['action'] ) ) {
```
Comment (matipojo) [inline includes/frontend.php]: Plugin::elementor()->preview->is_preview_mode()

File: includes/utils.php [hunk 4]
```diff
@@ -639,4 +639,11 @@ public static function get_meta_viewport( $context = '' ) {
 		 */
 		return apply_filters( 'elementor/template/viewport_tag', $meta_tag, $context );
 	}
+
+	public static function elementor_print_js_config( $js_var, $config, $already_jason = false ) {
```
Comment (matipojo) [inline includes/utils.php]: already_jason => already_json

**Review by matipojo — CHANGES_REQUESTED** [review]

File: includes/utils.php (hunk 4 above)
Comment (nuritsha) [reply includes/utils.php]: ho! sure

File: includes/frontend.php (hunk 3 above)
Comment (nuritsha) [reply includes/frontend.php]: like so?
```		
if ( Plugin::instance()->preview->is_preview_mode() || ( isset( $_REQUEST['action'] ) && 'elementor' === $_REQUEST['action'] ) ) {
			return str_replace( '<script', '<script data-cfasync="false"', $tag );
		}

		return $tag;
```

**Pull request merged**